### DISCUSSION TITLE
Determine view ID from within tree data provider

### DISCUSSION THREAD

**Discussion by gee4vee** [microsoft/vscode-discussions#2059] [Extension Development QnA] [answered]
I share a tree data provider across two views (so my custom tree view can reside in our custom view container and in the explorer container). When I click the refresh button in the tree view, is it possible to determine from the provider which view it came from so that I can show progress on that view?

**Comment by isidorn** [comment, marked as answer]
Hmm can you pass that as the argument to your data provider when you create it? So each data provider you have is aware where it is being used?

**Reply by gee4vee** [reply]
@isidorn The thing is I am using a single tree data provider instance for both views. Should I use a separate instance for each view?

**Reply by isidorn** [reply]
Just use separate instances for now imho.
